src/transformStateWithClones.js

```diff
@@ -7,7 +7,34 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const OriginalObj = Object.assign({}, state);
+  const res = [];
+
+  for (const action in actions) {
+    const operation = actions[action];
+
+    switch (operation.type) {
+      case 'addProperties':
+        Object.assign(OriginalObj, operation.extraData);
+        res.push(Object.assign({}, OriginalObj));
```

Comment:
you don't need to push originalObject to the result for each case. Try to move this after switch..case construction so that it is added to the result after each iteration of the loop. Please fix for all casses